client/src/components/activities/ActivityForm/RecurrenceForm/RecurrenceForm.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+	daysOfMonth,
+	daysOfWeek,
+	FREQUENCY,
+	frequencyOptions,
+	INTERVAL_UNIT
+} from "@/components/activities/ActivityForm/RecurrenceForm/constants";
+import DaySelector from "@/components/activities/ActivityForm/RecurrenceForm/DaySelector";
+import SharedStyle from "@/components/activities/ActivityForm/RecurrenceForm/style/DaySelector.style";
+import { Checkbox } from "@/components/utility/Checkbox/Checkbox";
+import { Tooltip } from "@mantine/core";
+import type { IntervalUnit } from "@shared/types/data/utility.types";
+import { LucideAlertCircle } from "lucide-react";
+import type useActivityForm from "../useActivityForm";
+import S from "./style/RecurrenceForm.style";
+
+export default function RecurrenceForm({
+	recurrence,
+	isRecurring,
+	intervalUnitSuffix,
+	toggleRecurring,
+	updateRecurrence,
+	setSelection,
+	resetSelection,
+	validRecurrence
+}: Pick<
+	ReturnType<typeof useActivityForm>,
+	| "recurrence"
+	| "isRecurring"
+	| "intervalUnitSuffix"
+	| "toggleRecurring"
+	| "updateRecurrence"
+	| "setSelection"
+	| "resetSelection"
+	| "validRecurrence"
+>) {
+	if (!isRecurring) {
+		return (
+			<S.Container>
+				<S.CheckboxWrapper style={{ maxWidth: "max-content" }}>
+					<span>Recurring activity?</span>
+					<Checkbox checked={isRecurring} onChange={toggleRecurring} />
+				</S.CheckboxWrapper>
+			</S.Container>
+		);
+	}
+
+	return (
+		<S.Container style={{ position: "relative" }}>
+			{!validRecurrence && (
+				<span style={{ position: "absolute", top: 0, right: 0 }}>
+					<Tooltip
+						label={
+							<div
```

Comment:
I am still not quite sure how I want to handle one-off cases like this. Maybe it should just be a styled `Tooltip.Wrap` component or something, even if it only ends up being used once or twice.